### PR TITLE
feat: [SFEQS-1352] Add analytics event

### PR DESCRIPTION
**Pull request by grausof**
Adds sending of analytics events to `datalake` eventhub (PDV).

#### List of Changes
Added the following analytics events:
-  Document uploaded 
- Document rejected
- Signature ready
- Signature send
- Notification rejected
- Signature signed
- Signature rejected
- QTSP errors

#### How Has This Been Tested?
TEsted locally

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-func-sign-issuer/src/app/use-cases/send-notification.ts [hunk 1]
```diff
@@ -78,7 +80,27 @@ export const makeSendNotification =
...
+              firstTaskEither,
+              TE.altW(() =>
+                pipe(
+                  signatureRequest,
+                  createAndSendAnalyticsEvent(EventName.NOTIFICATION_REJECTED),
+                  TE.chain(() => firstTaskEither)
+                )
+              )
+            )
+        ),
```
Comment (lucacavallaro) [inline apps/io-func-sign-issuer/src/app/use-cases/send-notification.ts:103]: it's not very linear, try to rewrite it using only one `createAndSendAnalyticsEvent` at the end, the only thing that changes is the event name

File: apps/io-func-sign-user/src/app/use-cases/validate-signature.ts [hunk 2]
```diff
@@ -105,15 +110,63 @@ export const makeValidateSignature =
...
+                pipe(
+                  firstTaskEither,
+                  TE.alt(() =>
+                    pipe(
+                      signatureRequest,
+                      createAndSendAnalyticsEvent(EventName.QTSP_API_ERROR),
+                      () => firstTaskEither
+                    )
+                  )
+                )
```
Comment (lucacavallaro) [inline apps/io-func-sign-user/src/app/use-cases/validate-signature.ts:124]: ditto (maybe here you can use `TE.chainFirst`)

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: apps/io-func-sign-user/src/app/use-cases/validate-signature.ts (hunk 2 above)
Comment (grausof) [reply apps/io-func-sign-user/src/app/use-cases/validate-signature.ts:124]: Anfortunately for `Alt` there is no `TE.altFirst` lik e`TE.chainFirst` so this is the only way to do it.

File: apps/io-func-sign-issuer/src/app/use-cases/send-notification.ts (hunk 1 above)
Comment (grausof) [reply apps/io-func-sign-issuer/src/app/use-cases/send-notification.ts:103]: it is not possible to do this as the first time is a chain while the second call is inside an Alt.

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**